File: CS/PreviewClickSample/DrillDownReport.cs
using System.Collections.Generic;
using System.Data;
using System.Drawing.Printing;
using System.Linq;
using DevExpress.XtraReports.UI;
// ...

namespace PreviewClickSample {
    public partial class DrillDownReport : DevExpress.XtraReports.UI.XtraReport {
        const string sShowDetail = "Show Detail";
        const string sHideDetail = "Hide Detail";

        public DrillDownReport() {
            InitializeComponent();
            expandedValuesParameter.Type = typeof(List<int>);
            expandedValuesParameter.Value = new List<int>();
            lbShowHide.BeforePrint += lbShowHide_BeforePrint;
            DetailReport.BeforePrint += DetailReport_BeforePrint;
        }

        List<int> ExpandedValues {
            get {
                return (List<int>)expandedValuesParameter.Value;
            }
        }

        bool ShouldShowDetail(int catID) {
            return ExpandedValues.Contains(catID);
        }

        private void lbShowHide_BeforePrint(object sender, PrintEventArgs e) {
            XRLabel label = (XRLabel)sender;
            if (ShouldShowDetail((int)label.Tag)) {
                label.Text = sHideDetail;
            }
            else {
                label.Text = sShowDetail;
            }
        }

        private void DetailReport_BeforePrint(object sender, PrintEventArgs e) {
            e.Cancel = !ShouldShowDetail((int)GetCurrentColumnValue("CategoryID"));
        }

        // The following code populates the XRChart with data.
        private void xrChart1_BeforePrint(object sender, PrintEventArgs e) {
            var chart = ((XRChart)sender);
            chart.DataSource = (from p in nwindDataSet1.Products
                                where p.CategoryID == (int)GetCurrentColumnValue("CategoryID")
                                select new {
                                    CategoryID = p.CategoryID,
                                    ProductName = p.ProductName,
                                    UnitsInStock = p.UnitsInStock,
                                    myRow = p.UnitPrice * p.UnitsInStock
                                }).ToArray();

            chart.Series[1].ArgumentDataMember = "ProductName";
            chart.Series[1].ValueDataMembers.AddRange(new string[] { "myRow" });

            chart.Series[0].ArgumentDataMember = "ProductName";
            chart.Series[0].ValueDataMembers.AddRange(new string[] { "UnitsInStock" });
        }
    }
}

File: VB/PreviewClickSample/DrillDownReport.vb
Imports Microsoft.VisualBasic
Imports System.Collections.Generic
Imports System.Data
Imports System.Drawing.Printing
Imports System.Linq
Imports DevExpress.XtraReports.UI
' ...

Namespace PreviewClickSample
	Partial Public Class DrillDownReport
		Inherits DevExpress.XtraReports.UI.XtraReport
		Private Const sShowDetail As String = "Show Detail"
		Private Const sHideDetail As String = "Hide Detail"

		Public Sub New()
			InitializeComponent()
			expandedValuesParameter.Type = GetType(List(Of Integer))
			expandedValuesParameter.Value = New List(Of Integer)()
			AddHandler lbShowHide.BeforePrint, AddressOf lbShowHide_BeforePrint
			AddHandler DetailReport.BeforePrint, AddressOf DetailReport_BeforePrint
		End Sub

		Private ReadOnly Property ExpandedValues() As List(Of Integer)
			Get
				Return CType(expandedValuesParameter.Value, List(Of Integer))
			End Get
		End Property

		Private Function ShouldShowDetail(ByVal catID As Integer) As Boolean
			Return ExpandedValues.Contains(catID)
		End Function

		Private Sub lbShowHide_BeforePrint(ByVal sender As Object, ByVal e As PrintEventArgs)
			Dim label As XRLabel = CType(sender, XRLabel)
			If ShouldShowDetail(CInt(Fix(label.Tag))) Then
				label.Text = sHideDetail
			Else
				label.Text = sShowDetail
			End If
		End Sub

		Private Sub DetailReport_BeforePrint(ByVal sender As Object, ByVal e As PrintEventArgs)
			e.Cancel = Not ShouldShowDetail(CInt(Fix(GetCurrentColumnValue("CategoryID"))))
		End Sub

		' The following code populates the XRChart with data.
		Private Sub xrChart1_BeforePrint(ByVal sender As Object, ByVal e As PrintEventArgs) Handles xrChart1.BeforePrint
			Dim chart = CType(sender, XRChart)
			Dim categoryId = CInt(Fix(GetCurrentColumnValue("CategoryID")))
            chart.DataSource = ( _
              From p In nwindDataSet1.Products _
              Where p.CategoryID = categoryId Select New With {Key .CategoryID = p.CategoryID, Key .ProductName = p.ProductName, _
                                                               Key .UnitsInStock = p.UnitsInStock, Key .myRow = p.UnitPrice * p.UnitsInStock}).ToArray()

			chart.Series(1).ArgumentDataMember = "ProductName"
			chart.Series(1).ValueDataMembers.AddRange(New String() { "myRow" })

			chart.Series(0).ArgumentDataMember = "ProductName"
			chart.Series(0).ValueDataMembers.AddRange(New String() { "UnitsInStock" })
		End Sub
	End Class
End Namespace
